client/dashboard/sites/settings-transfer-site/invitation-email-sent.tsx

```diff
@@ -0,0 +1,93 @@
+import { useMutation } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
+import { ExternalLink, Spinner } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect, useState } from 'react';
+import { siteOwnerTransferConfirmMutation } from '../../app/queries';
+import Notice from '../../components/notice';
+import type { SiteTransferConfirmation } from '../../data/types';
+
+export function InvitationEmailSent( {
+	siteSlug,
+	confirmationHash,
+}: {
+	siteSlug: string;
+	confirmationHash: string;
+} ) {
+	const [ newOwnerEmail, setNewOwnerEmail ] = useState( '' );
+	const [ hasError, setHasError ] = useState( false );
+	const mutation = useMutation( siteOwnerTransferConfirmMutation( siteSlug ) );
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const router = useRouter();
+
+	// The page is accessed via the confirmation email, so this is the only place where the request can be triggered.
+	// It would be better if
+	// * we have the server trigger the mutation before rendering the page
+	// * we first show a page, clarifying what is going to happen with an "accept" button or something.
+	useEffect( () => {
+		mutation.mutate(
+			{ hash: confirmationHash },
+			{
+				onSuccess: ( { transfer, new_owner_email }: SiteTransferConfirmation ) => {
+					if ( transfer ) {
+						createSuccessNotice(
+							sprintf(
+								/* translators: %(newOwnerEmail)s - the new owner's email */
+								__( 'The site has been successfully transferred to %(newOwnerEmail)s.' ),
+								{
+									newOwnerEmail: new_owner_email,
+								}
+							),
+							{ type: 'snackbar' }
+						);
+						router.navigate( { to: '/sites' } );
+						return;
+					}
+
+					setNewOwnerEmail( new_owner_email );
+				},
```

Comment:
Does it really work? I always get 404 because I'm no longer the owner, thus can't access the page anymore...

<img width="1023" alt="image" src="https://github.com/user-attachments/assets/359760af-ab8d-457b-bbd6-aeab964f3cc7" />

Reply:
Maybe we should invalidate the query here instead of doing it in the mutation itself 🤔

Reply:
I'm not sure what you mean 🤔 

I'm thinking we should update the condition here

https://github.com/Automattic/wp-calypso/blob/cfa05568e60129aa19754ffdcf1dacec3594854d/client/dashboard/sites/settings-transfer-site/index.tsx#L25-L27

to this

```js
	if ( ! canTransferSite && ! confirmationHash ) {
		throw notFound();
	}
```

Reply:
Sorry for not being clear earlier. The `transfer: true` flag is set only when ownership is transferred to an existing member of the site. However, it seems a bit inconsistent that in some cases, transfer might still not be true...

You saw the 404 page because the transfer was processed via email. This happens when the transfer is completed and the page attempts to refetch the site data at a certain point.

Reply:
Ohh... I see now! OK, tested both cases and they work.

I think we should revert [my suggestion](https://github.com/Automattic/wp-calypso/pull/103700#discussion_r2110933157), I misunderstood something. We should throw 404 otherwise we keep hitting the confirmation endpoint.